hamt/hamt.go

```diff
@@ -188,26 +193,14 @@ func (ds *Shard) Node() (ipld.Node, error) {
 	return out, nil
 }
 
-type shardValue struct {
-	key string
-	val *ipld.Link
-}
-
-// Link returns a link to this node
-func (sv *shardValue) Link() (*ipld.Link, error) {
-	return sv.val, nil
-}
+func (ds *Shard) makeShardValue(lnk *ipld.Link) *Shard {
+	lnk2 := *lnk
+	s, _ := makeShard(ds.dserv, ds.tableSize)
```

Comment:
@schomatis  
Did you mean that we should add `panic` here?  
IMO, I think the `err` always will be `nil` , because the `ds.tableSize` is a correct `size`.

Reply:
I would add an error check and then have makeShardValue be able to return an error, then add error checks on the two places it's called, which are both in functions which already pass on errors.

Reply:
@hannahhoward yeah, a good advise

Reply:
@hannahhoward @schomatis fixed